ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/DynamicPivotBlockManager.java

```diff
@@ -47,20 +43,18 @@
   private Optional<BlockHeader> lastBlockFound;
 
   public DynamicPivotBlockManager(
-      final WorldDownloadState<REQUEST> worldDownloadState,
       final FastSyncActions fastSyncActions,
       final SnapSyncState fastSyncState,
       final int pivotBlockWindowValidity,
       final int pivotBlockDistanceBeforeCaching) {
-    this.worldDownloadState = worldDownloadState;
     this.syncActions = fastSyncActions;
     this.syncState = fastSyncState;
     this.pivotBlockWindowValidity = pivotBlockWindowValidity;
     this.pivotBlockDistanceBeforeCaching = pivotBlockDistanceBeforeCaching;
     this.lastBlockFound = Optional.empty();
   }
 
-  public void check(final Consumer<BlockHeader> onNewPivotBlock) {
+  public void check(final BiConsumer<BlockHeader, Boolean> onNewPivotBlock) {
     syncState
         .getPivotBlockNumber()
         .ifPresent(
```

Comment:
Optimization `final long currentPivotBlockNumber = syncState.getPivotBlockNumber().orElseThrow();` could be removed since the value is already in `blockNumber` variable